src/components/guestsList/guestsList.js

```diff
@@ -0,0 +1,243 @@
+    const { numberOfEntries, pending, listData } = this.props;
+
+    const bottom = target.scrollHeight - target.scrollTop === target.clientHeight;
+
+    if (numberOfEntries > (currentPage + 1) * limit && bottom && !pending) {
+      const newPage = currentPage + 1;
+      const updatedData = [...previousData, ...(listData || [])];
+
+      this.setState({
+        previousData: updatedData,
+        currentPage: newPage
+      });
+    }
+  };
+
+  renderLoader() {
+    const { currentPage } = this.state;
+    const { filterGuestsData, listData, pending } = this.props;
+
+    if (currentPage > 0 && pending) {
+      const scrollLoader = (
+        <Loader
+          variant="table"
+          tableProps={{
+            borders: false,
+            colCount: filterGuestsData?.length || (listData?.[0] && Object.keys(listData[0]).length) || 1,
+            rowCount: 0,
+            variant: TableVariant.compact
+          }}
+        />
+      );
+
+      return <div className="curiosity-table-scroll-loader__custom">{scrollLoader}</div>;
+    }
+
+    return null;
+  }
+
+  /**
+   * Render a guests table.
+   *
+   * @returns {Node}
+   */
+  renderTable() {
+    const { previousData } = this.state;
+    const { filterGuestsData, listData } = this.props;
+    let updatedColumnHeaders = [];
+
+    const updatedRows = [...previousData, ...(listData || [])].map(({ ...cellData }) => {
+      const { columnHeaders, cells } = inventoryListHelpers.parseRowCellsListData({
+        filters: filterGuestsData,
+        cellData
+      });
+
+      updatedColumnHeaders = columnHeaders;
+
+      return {
+        cells
+      };
+    });
+
+    return (
+      <div className="curiosity-table-scroll" style={{ height: `200px` }}>
+        <div className="curiosity-table-scroll-list" onScroll={this.onScroll}>
+          {this.renderLoader()}
+          {(updatedRows.length && (
+            <Table
+              borders={false}
+              variant={TableVariant.compact}
+              className="curiosity-inventory-list"
+              columnHeaders={updatedColumnHeaders}
+              rows={updatedRows}
+            />
+          )) ||
+            null}
+        </div>
+      </div>
+    );
+  }
+
+  /**
+   * Render a guest list table.
+   *
+   * @returns {Node}
+   */
+  render() {
+    const { currentPage } = this.state;
+    const { error, filterGuestsData, listData, pending, perPageDefault } = this.props;
+
+    return (
+      <div className={`fadein ${(error && 'blur') || ''}`}>
+        {pending && currentPage === 0 && (
+          <Loader
+            variant="table"
+            tableProps={{
+              borders: false,
+              colCount: filterGuestsData?.length || (listData?.[0] && Object.keys(listData[0]).length) || 1,
+              rowCount: perPageDefault,
+              variant: TableVariant.compact
+            }}
+          />
+        )}
+        {((!pending && currentPage === 0) || currentPage > 0) && this.renderTable()}
```

Comment:
Required:
Investigate changing this condition to simply `!pending`

Reply:
So this is subtle, but this it is required because of the "sub loading" display that happens on subsequent pages above "0"